src/libs/blueprint/conduit_blueprint_mpi_mesh.cpp

```diff
@@ -237,6 +237,69 @@ number_of_domains(const conduit::Node &n,
     return global_num_domains;
 }
 
+//-----------------------------------------------------------------------------
+void
+state(const conduit::Node &mesh,
+      conduit::Node &state,
+      MPI_Comm comm)
+{
+    Node local_state;
+    ::conduit::blueprint::mesh::state(mesh, local_state);
+
+
+    Node global_state_nodes;
+    relay::mpi::all_gather_using_schema(local_state,
```

Comment:
we could find the first instance of state and broadcast that to all, vs an all gather, which will copy all copies of state to all mpi tasks